### PR TITLE
add gem rails_utils

### PR DESCRIPTION
**Pull request by sunkai612**
for #14

This Pull Request sets up [rails_utils](https://github.com/winston/rails_utils).

### CODE REVIEW

**Comment by JuanitoFatas** [comment]
Could you look into `#page_class` & `#javascript_initialization` & `#flash_messages` and put them into corresponding places? Thanks!

**Comment by sunkai612** [comment]
Should I change `application.html` from `erb` to `slim` ?

**Comment by JuanitoFatas** [comment]
> Should I change application.html from erb to slim ?

Nope! Already done in #20. You need to rebase to pick up changes from #20. :blush:

**Comment by sunkai612** [comment]
just add `#page_class`, `#javascript_initialization` & `#flash_messages`

but not sure about should `javascript_include_tag 'application', 'data-turbolinks-track' => true` been removed or not.. :no_mouth:

**Comment by sunkai612** [comment]
Mayday Mayday, requesting code review help :astonished:

**Comment by JuanitoFatas** [comment]
> but not sure about should javascript_include_tag 'application', 'data-turbolinks-track' => true been removed or not..

Why you think this line should be removed? :information_desk_person:

**Comment by sunkai612** [comment]
because it looks like `#javascript_initialization` is going to generate custom script for each controller & action.

and that can be used with:
(1) including js on separate page as part;
(2) including js on `application.html` as all
both seems reasonable for me, which confused me :astonished:

**Comment by azi** [comment]
I thought  should be keep
1. javascript_include_tag 'application'  for all website common js function 
2. #javascript_initialization  for each controller specially function use

**Comment by JuanitoFatas** [comment]
> I thought should be keep
> 
> javascript_include_tag 'application' for all website common js function
> #javascript_initialization for each controller specially function use

:clap: :clap: :clap: :clap: Exactly.

**Comment by JuanitoFatas** [comment]
:shipit: (<--- Ship it, [why squirrel?](https://www.quora.com/GitHub/What-is-the-significance-of-the-Ship-It-squirrel))

**Comment by sunkai612** [comment]
@azi @JuanitoFatas thanks for explaining, very helpful :smile:

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -5,5 +5,7 @@ html
     = stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track' => true
     = javascript_include_tag 'application', 'data-turbolinks-track' => true
     = csrf_meta_tags
-  body
+  body class=page_title
```
Comment (AnNOtis) [inline app/views/layouts/application.html.slim:8]: I'm not sure. Shouldn't it be `page_class`?

Comment (JuanitoFatas) [inline app/views/layouts/application.html.slim:8]: @AnNOtis Could you open a new Pull Request to fix this? Thanks and good catch :+1:

Comment (AnNOtis) [inline app/views/layouts/application.html.slim:8]: Fixed in #25

**Pull request merged**